packages/plugin-ext/src/main/browser/comments/comment-thread-widget.tsx

```diff
@@ -498,6 +499,16 @@ export class ReviewComment<P extends ReviewComment.Props = ReviewComment.Props>
             </div>
         </div>;
     }
+    localeDate(stamp: string | undefined): string {
+        if (stamp === undefined) {
+            return '';
+        }
+        const date = Date.parse(stamp);
+        if (date && !Number.isNaN(date)) {
+            return new Date(date).toLocaleString();
```

Comment:
Hi, why is it need to create two `Date` instances for one string? Thanks!

Reply:
Hi, thanks for your very fast review @kittaakos! 
I am not sure to understand your comment: I only create 1 Date instance line 508, result of line 506 is a number.

Reply:
You're right, but you parse the string twice: `parse` and the `Date` constructor. Doing it once would be sufficient. What do you think?

From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date):

> When parsing date strings with the Date constructor (and Date.parse, they are equivalent)

```ts
parse(s: string | undefined): string {
  if (s) {
    const d = new Date(s);
    if (!Number.isNaN(d.getTime())) {
      return d.toLocaleString();
    }
  }
  return '';
}
```

Reply:
@kittaakos, thanks a lot for your explanation! I did simplify indeed the code with 5d615cd.

Reply:
Thank you so much!

Reply:
I appreciate your taking the time to update your PR after my comment. The code is correct, but I could not resist adding that the `date` check in the ` if (date && !isNaN(date.getTime())) {` condition is superfluous.

A constructor function [can return with `undefined`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor):

> The constructor method may have a return value. While the base class may return anything from its constructor, the derived class must return an object or undefined, or a [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) will be thrown.

But you will never get `undefined` after successfully calling it with `new` at runtime. See [here](https://github.com/microsoft/TypeScript/issues/29893#issuecomment-463409560). So you can eliminate the `date` check, which will always be `true`, and simplify the code if you wish.

I am happy to receive a correction if I am proven wrong on this. Thanks!

Reply:
Thanks @kittaakos, you are absolutely right! The code has been simplified in 6895086.